phpstan.neon.dist

```diff
@@ -0,0 +1,43 @@
+# PHPStan configuration for WordPress Core.
+#
+# To overload this configuration, copy this file to phpstan.neon and adjust as needed.
+#
+# https://phpstan.org/config-reference
+
+includes:
+  # The WordPress Core configuration file includes the base configuration for the WordPress codebase.
+  - tests/phpstan/base.neon
+  # The baseline file includes preexisting errors in the codebase that should be ignored.
+  # https://phpstan.org/user-guide/baseline
+  - tests/phpstan/baseline/level-0.php
+  - tests/phpstan/baseline/level-1.php
+  - tests/phpstan/baseline/level-2.php
+  - tests/phpstan/baseline/level-3.php
+  - tests/phpstan/baseline/level-4.php
+  - tests/phpstan/baseline/level-5.php
+  - tests/phpstan/baseline/level-6.php
+
+parameters:
+  level: 6
+  reportUnmatchedIgnoredErrors: true
```

Comment:
This is enabled for the sake of this PR, but we might want to turn it off before merging.

Ref: https://phpstan.org/user-guide/backward-compatibility-promise